core/extensions/extension.hpp

```diff
@@ -20,87 +22,195 @@ namespace kagome::extensions {
         size_t state, size_t func_index)>;
 
    public:
-    /// storage extensions
-    virtual uint8_t *ext_child_storage_root(const uint8_t *storage_key_data,
-                                            uint32_t storage_key_length,
-                                            uint32_t *written) = 0;
+    // -------------------------Storage extensions--------------------------
 
-    virtual void ext_clear_child_storage(const uint8_t *storage_key_data,
-                                         uint32_t storage_key_length,
-                                         const uint8_t *key_data,
-                                         uint32_t key_length) = 0;
+    /**
+     * @brief Deletes values by keys containing given prefix
+     * @param prefix_data pointer to the prefix
+     * @param prefix_length lemgth of the prefix
+     */
+    virtual void ext_clear_prefix(runtime::WasmPointer prefix_data,
```

Comment:
using does not introduce new type. Are you sure you want WasmPointer to behave like uint32_t?

Reply:
Yes, some other implementations (e.g. polkadot js [pointer type](https://github.com/polkadot-js/client/blob/master/packages/client-runtime/src/types.ts#L11)) use the same approach and it works well. Basically WasmPointer is just an offset to the WasmMemory which stores an vector of bytes